ParamHash should be combining parameters with a comma

jebw opened the pull request:
I'm not entirely convinced this is correct so it could do with some review _but_ param hash currently combines multiple parameters of the same type using a space so for example

add_output_parameter :vcodec, 'libvpx'
add_output_parameter :acodec, 'libvorbis'
add_output_parameter :vf, 'vflip'
add_output_parameter :vf, 'hflip'

Would result in

-vcodec libvpx -acodec libvorbis -vf vflip hflip

Surely this should instead be

-vcodec libvpx -acodec libvorbis -vf vflip,hflip

Is there any situation where use of the same parameter multiple times should result in the parameters being separated by spaces? If not then I think this should be probably merged?

jebw committed:
ParamHash should be combining parameters with a comma

attenzione commented:
i think it's also incorrect, it should be 
`value = value.join(',') if value.is_a?(Array)`
because if value is `Number` it will be joined, t.i is incorrect

jebw commented:
Yes, that would make more sense - I'll amend the branch

jebw commented:
I've updated the branch and the tests.

The tests were explicitly checkout for '-parameter first second third' in the output so I've updated them but can any one comment if there's a valid scenario where '-parameter first second third' would be the desired output - how consistent is ffmpeg in this regard?

attenzione commented:
completely identical to avconv
`ffmpeg -i input.mp4 -vf "hflip,vflip,format=yuv420p" -metadata:s:v rotate=0 -codec:v libx264 -codec:a copy output.mkv`
btw, in a row above, metadata about rotation is stripped, this should be somehow implemented.
i have years ago issue, when used paperclip-ffmpeg and apply rotation for original file, without removing metadata from it. So other styles was rotated again (twice from original).
so in other gem - av-transcoder, if auto_rotate is true, don't forget to remove metadata. But rotation not yet implemented there...

owahab commented:
Instead of removing metadata, why not just update it?

attenzione commented:
when rotation is not required, there is no metadata about rotation at all.
